components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java

```diff
@@ -0,0 +1,116 @@
+
+/**
+ * Alphanumeric validator.
+ */
+public class AlphanumericValidator extends AbstractRulesValidator {
+
+    private final List<String> allowedFields = new ArrayList<String>() {{
+        add(USERNAME);
+    }};
+
+    @Override
+    public boolean isAllowedField(String field) {
+
+        return allowedFields.contains(field);
+    }
+
+    /**
+     * Validate the string against the validation criteria.
+     *
+     * @param context   Validation Context.
+     * @return boolean
+     * @throws InputValidationMgtClientException Error when string does not satisfy the validation criteria
+     */
+    @Override
+    public boolean validate(ValidationContext context) throws InputValidationMgtClientException {
+
+        String value = context.getValue();
+        String field = context.getField();
+        Map<String, String> attributesMap = context.getProperties();
+        String alphanumericRegEx = DEFAULT_ALPHANUMERIC_REGEX_PATTERN;
+
+        // Check whether value satisfies the alphanumeric criteria.
+        if (attributesMap.containsKey(ENABLE)) {
+            if (Boolean.parseBoolean(attributesMap.get(ENABLE)) && value != null && !value.matches(alphanumericRegEx)) {
+                throw new InputValidationMgtClientException(ERROR_VALIDATION_ALPHANUMERIC_FORMAT_MISMATCH.getCode(),
+                    ERROR_VALIDATION_ALPHANUMERIC_FORMAT_MISMATCH.getMessage(), String.format(
+                    ERROR_VALIDATION_ALPHANUMERIC_FORMAT_MISMATCH.getDescription(), field, alphanumericRegEx));
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Get list of supported properties for the validator.
+     *
+     * @return  List<Property>
+     */
+    @Override
+    public List<Property> getConfigurationProperties() {
+
+        List<Property> configProperties = new ArrayList<>();
+        int parameterCount = 0;
+
+        Property enable = new Property();
+        enable.setName(ENABLE);
+        enable.setDisplayName("Alphanumeric field value");
+        enable.setDescription("Validate whether the field value is alphanumeric.");
+        enable.setType("boolean");
+        enable.setDisplayOrder(++parameterCount);
+        configProperties.add(enable);
+
+        return configProperties;
+    }
+
+
+    /**
+     * Validate the configuration values of the properties for the validator.
+     *
+     * @param context   Validation Context.
+     * @return  boolean
+     */
+    @Override
+    public boolean validateProps(ValidationContext context) throws InputValidationMgtClientException {
+
+        Map<String, String> properties = context.getProperties();
+        validatePropertyName(properties, this.getClass().getSimpleName(), context.getTenantDomain());
+        if (properties.get(ENABLE) != null && !validateBoolean(properties.get(ENABLE),
+                ENABLE, context.getTenantDomain())) {
+            properties.remove(ENABLE);
```

Comment:
This should be an error scenario

Reply:
Addressed with commit 6e4ccba6376101562c801f735b660c94f4a280b2